Timeout removing messages

nullpixel opened the pull request:
- Fixes a bug where some messages would not be removed
- Times out deleting messages (dabbit request)
- Wait for the bot's message to be removed before removing the user's message

nullpixel committed:
Merge pull request #1 from SteamingMutt/uservoice

Keep up to date wew

nullpixel committed:
No emebd pls

nullpixel committed:
Removed that annoying space

nullpixel committed:
Dabbit request: timeout removing messages

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -319,23 +319,27 @@ commands['approve'] = {
-            message.reply('You cannot approve your own submission.').then(deletThis)
-            message.delete()
+            message.reply('You cannot approve your own submission.').then(delay(15000)).then((deletThis) => {
+              messge.delete()
```
messge

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -319,23 +319,27 @@ commands['approve'] = {
-            message.reply('You need to be an admin to approve this report.').then(deletThis)
-            message.delete()
+            message.reply('You need to be an admin to approve this report.').then(delay(15000)).then((deletThis) => {
+              messge.delete()
```
messge

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -319,23 +319,27 @@ commands['approve'] = {
-            message.reply('this report has already been closed.').then(deletThis)
-            message.delete()
+            message.reply('this report has already been closed.').then(delay(15000)).then((deletThis) => {
+              messge.delete()
```
messge

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -319,23 +319,27 @@ commands['approve'] = {
-          message.reply(`You've successfully submitted your approval for this report.`).then(deletThis)
-          message.delete()
+          message.reply(`You've successfully submitted your approval for this report.`).then(delay(15000)).then((deletThis) => {
+            messge.delete()
```
messge

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -357,31 +361,36 @@ commands['deny'] = {
-      message.reply('You need to supply a reason to deny this report.').then(deletThis)
-      message.delete()
+      message.reply('You need to supply a reason to deny this report.').then(delay(15000)).then((deletThis) => {
+        messge.delete()
```
messge

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -357,31 +361,36 @@ commands['deny'] = {
-        message.reply('No report was found with this ID').then(deletThis)
-        message.delete()
+        message.reply('No report was found with this ID').then(delay(15000)).then((deletThis) => {
+          messge.delete()
```
messge

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -357,31 +361,36 @@ commands['deny'] = {
-            message.reply('You need to be an admin to deny this report.').then(deletThis)
-            message.delete()
+            message.reply('You need to be an admin to deny this report.').then(delay(15000)).then((deletThis) => {
+              messge.delete()
```
messge

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -357,31 +361,36 @@ commands['deny'] = {
-            message.reply('this report has already been closed.').then(deletThis)
-            message.delete()
+            message.reply('this report has already been closed.').then(delay(15000)).then((deletThis) => {
+              messge.delete()
```
messge

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -357,31 +361,36 @@ commands['deny'] = {
-          message.reply(`You've successfully submitted your denial for this report.`).then(deletThis)
-          message.delete()
+          message.reply(`You've successfully submitted your denial for this report.`).then(delay(15000)).then((deletThis) => {
+            messge.delete()
```
messge

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -420,8 +429,9 @@ commands['submit'] = {
-        message.channel.sendMessage(['Thank you for your feedback!\nWe\'ve send it to the custodians for review!']).then(deletThis)
-        message.delete()
+        message.channel.sendMessage(['Thank you for your feedback!\nWe\'ve send it to the custodians for review!']).then(delay(15000)).then((deletThis) => {
+          messge.delete()
```
messge

nullpixel1 commented:
dammit, fixing

nullpixel committed:
:face_palm: I'm pretty stupid

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -319,23 +319,27 @@ commands['approve'] = {
           if (state[content[0]].sudo && r !== 2) {
-            message.reply('You need to be an admin to approve this report.').then(deletThis)
-            message.delete()
+            message.reply('You need to be an admin to approve this report.').then(delay(15000)).then((deletThis) => {
```
This doesn't work, this way you're passing the function as an argument to the anonymous function to the chain.

```js
.then((msg) => {
  message.delete()
  deletThis(msg)
})
```

nullpixel replied:
oh, right. didn't think of that. thanks 👍

nullpixel committed:
Fix the undefined delay issue

nullpixel1 commented:
@SteamingMutt Should all be fixed now :)

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -570,6 +586,14 @@ function wait(bot, msg) {
 }
 
+function delay(delayMS) {
+  return function(arg){
```
Why return 2 function statements if you don't do anything with it? Just return the promise.

nullpixel committed:
Only return a promise

Arzte commented on Commands/Commands/utils.js:
```diff
@@ -319,23 +319,31 @@ commands['approve'] = {
           if (state[content[0]].sudo && r !== 2) {
-            message.reply('You need to be an admin to approve this report.').then(deletThis)
-            message.delete()
+            message.reply('You need to be an admin to approve this report.').then(delay(15000)).then((msg) => {
```
Can you make the ``delay()`` time into a config option?

nullpixel replied:
Sure; was debating doing this out of the box, but now you asked I can

nullpixel committed:
Added config option for delay time

nullpixel committed:
Missed a few

Dougley commented on Commands/Commands/utils.js:
```diff
@@ -570,6 +592,12 @@ function wait(bot, msg) {
 
+function delay(delayMS) {
+  return new Promise((resolve) => {
+    setTimeout(() => resolve(arg), delayMS);
```
`arg` is undefined.

nullpixel replied:
Whoops, that's because I removed the function return. I'm trying to write this code blind, as I haven't got my own instance up yet

nullpixel committed:
Fix undefined arg

I'm sorry for the hastle and I will get my own instance up so I can test it next time

Arzte committed:
Rename deletThis() to deleteThis()

@dougly needs to write better function names *cough* *cough*

nullpixel1 commented:
I'm setting my own instance up right now, so I can actually see my stupid mistakes, it's hard to code blind :^3

SirDoctors commented:
Unrelated to the PR: @dougly I'd like to apolgize for pinging you in the commit, I meant to ping @SteamingMutt. Truly sorry to waste your time

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

SirDoctors commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Remove bot messages after a set interval](https://trello.com/c/CMNyuNHO/11-remove-bot-messages-after-a-set-interval)

Dougly commented:
😅
